### ISSUE TITLE
insecure HTTP whitelist

### ISSUE THREAD

**Issue by soratobukuroneko** [tobykurien/WebApps#16]
Would it be possible to have some whitelisting system for HTTP like the one for 3rd party requests? Thx

**Comment by tobykurien** [comment]
It's possible, yes. 

My personal view is that it's a bad idea. You may forget about the http unblocking you added, and then hop onto public wifi, for example. There is an option to disable the HTTP blocking for those cases where you absolutely need it, but hopefully you'll only enable that on a secure network. 

I feel that the core value proposition of WebApps is that it absolutely prevents insecure connections by default. It is the single reason I wrote the app.

**Comment by tobykurien** [comment]
BTW, duplicate of https://github.com/tobykurien/WebApps/issues/16 - let's discuss there if you feel that there is good reason to allow HTTP.

**Comment by AltNico** [comment]
@tobykurien Thank you for your response. I understand why you choose to force HTTPS but the consequent result would be to remove the option to allow HTTP totally. I see no reason why 3rd party requests are less important than the main request, as in my web app for example the important data comes from 3rd party requests (CORS REST API).

It's the same like you mentioned above: maybe you forget about this preference and hop into an open WiFi. But as an user, you should be able o decide on your own whether you want to take that risk. Everything else would be paternalism like Apple does, IMHO.

**Comment by tobykurien** [comment]
@AltNico the reason why 3rd party HTTP was an option is that back when I first released the app, most CDNs didn't support HTTPS. The main website is usually the one with the auth cookies, etc. which is why I figured 3rd party requests were less risky (also refer and 3rd party cookies are disabled). However, it might now be the time to remove that option entirely.

As for forcing things onto the user (a la Apple) - this app is all about privacy, so it makes sense to remove options that compromise that. Users are always welcome to use a normal browser (e.g. Lightning) to access sites insecurely. I also feel that by enforcing encryption, users will be more likely to bug insecure website owners to properly secure their site, and make it more obvious to users which sites are insecure.
